### PR TITLE
🌱 Use setup-envtest from CR main (05/2024) to use envtest binaries from CT releases

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Context: https://github.com/kubernetes-sigs/controller-runtime/pull/2811

I'm going to backport this to all branches (and also CAPV)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,10 +101,10 @@ KUSTOMIZE_BIN := kustomize
 KUSTOMIZE := $(abspath $(TOOLS_BIN_DIR)/$(KUSTOMIZE_BIN)-$(KUSTOMIZE_VER))
 KUSTOMIZE_PKG := sigs.k8s.io/kustomize/kustomize/v5
 
-SETUP_ENVTEST_VER := v0.0.0-20240215143116-d0396a3d6f9f
+SETUP_ENVTEST_VER := v0.0.0-20240506183901-d950bb92291d
 SETUP_ENVTEST_BIN := setup-envtest
 SETUP_ENVTEST := $(abspath $(TOOLS_BIN_DIR)/$(SETUP_ENVTEST_BIN)-$(SETUP_ENVTEST_VER))
-SETUP_ENVTEST_PKG := sigs.k8s.io/controller-runtime/tools/setup-envtest
+SETUP_ENVTEST_PKG := github.com/sbueringer/controller-runtime/tools/setup-envtest
```
Comment (sbueringer) [inline Makefile:107]: Using the version from my PR until it is merged. Going to change this to latest commit on CR main before this PR here is merged

File: hack/go-tools-build.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+# Create TMP_MODULE_DIR to create a go module for building the binary.
+rm -r "${TMP_MODULE_DIR}" || true
+mkdir -p "${TMP_MODULE_DIR}"
+cd "${TMP_MODULE_DIR}"
+
+# Initialize a go module and place a tools.go file for building the binary.
+go mod init "tools"
+# Set require for "sigs.k8s.io/cluster-api" to let go resolve the tools version via the CAPI version.
+go mod edit -replace "sigs.k8s.io/controller-runtime/tools/setup-envtest=github.com/sbueringer/controller-runtime/tools/setup-envtest@v0.0.0-20240506183901-d950bb92291d"
```
Comment (sbueringer) [inline hack/go-tools-build.sh:53]: Using the version from my PR until it is merged. Going to change this to latest commit on CR main before this PR here is merged

@chrischdi Thx for the script in CAPV btw that I copied almost all of this from :)

**Comment by sbueringer** [comment]
cc @fabriziopandini

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**